examples/gym/tf_model.py

```diff
@@ -21,3 +25,61 @@ def policy_model(obs_dim: int,
 
     return tf.keras.Model([obs], [value_logit, pi_logit], name="policy_model")
 
+def policy_model_cnn(obs_type: str = "RGB",
+                     obs_dim: tuple = (84, 84, 4),
+                     hidden_layers: int = 1,
+                     conv_layers: int = 2,
+                     filters_dim: list = [32, 64],
+                     kernel_dim: list = [8, 4],
+                     strides_dim: list = [4, 3],
+                     hidden_dim: int = 1024,
+                     activation: str = "relu") -> tf.keras.Model:
+
+    obs = layers.Input(shape=obs_dim, dtype=tf.float32, name="obs")
+    if obs_type == "RGB":
+        x = layers.Lambda(lambda layer: layer / 255)(obs) # normalize by 255, if necessary
+
+    # Convolutions on the frames on the screen
+    x = layers.Conv2D(filters_dim[0], (kernel_dim[0],kernel_dim[0]), strides_dim[0], activation=activation)(x)
+    for i in range(conv_layers-1):
+        x = layers.Conv2D(filters_dim[i], (kernel_dim[i],kernel_dim[i]), strides_dim[i], activation=activation)(x)
+    x = layers.Flatten()(x)
+
+    for _ in range(hidden_layers):
+        x = layers.Dense(hidden_dim, activation=activation)(x)
+    
+    value_logit = layers.Dense(1, name="value")(x)
+    pi_logit = layers.Dense(1, name="prior")(x)
+
+    return tf.keras.Model([obs], [value_logit, pi_logit], name="policy_model_cnn")
+
+
+
+def policy_model_2(obs_dim: Tuple[int],
```

Comment:
any reason for the two models? was the first one just for testing to see if `steps` is needed as an input?

Reply:
oh, actually, this might still be unresolved.  we have this issue around the state representation non-uniqueness leading to cycles.  the question is how to address this -- in the hallway example, the steps were already part of the state obs.  but in gridworld the obs is just the array, so i added steps here to enforce DAG structure.  i think in general the get_obs method implemented by a user could lead to cycles